### PR TITLE
Extension Wiki generator.

### PR DESCRIPTION
**Pull request by phanirithvij**
I've added a file `scripts/genwiki.sh` when executed from the root directory i.e. `tachiyomi-extensions/`
as 
```shell
# cd tachiyomi-extensions/
bash scripts/genwiki.sh
```
it generates a file `Extensions.md` which can be used to copy/paste into the wiki page directly or cloning the wiki locally and using it.

**NOTE** needs to be executed and updated each and every time a new release of **any** extension is made.

### CODE REVIEW

**Comment by phanirithvij** [comment]
@CarlosEsco, please look have a look.

**Comment by Presac** [comment]
It makes quite a lot of empty space. It feels like a list like this should be a bit more compact. As it is right now, I can only see 7 sources at once.

**Comment by phanirithvij** [comment]
@Presac

> It makes quite a lot of empty space. It feels like a list like this should be a bit more compact. As it is right now, I can only see 7 sources at once.

I'll add a details tag then.
Check after a few min.

**Comment by einstein95** [comment]
Made it so it makes 1 table per language and the resulting HTML looks cleaner. ALso tidied up spacing in the script as it was inconsistent.
[genwiki.sh.zip](https://github.com/inorichi/tachiyomi-extensions/files/2897973/genwiki.sh.zip)

**Comment by phanirithvij** [comment]
@Presac @einstein95 please check again.

**Comment by Presac** [comment]
Looks a lot cleaner now. It's a nice touch you can minimize each section/language.

**Comment by arkon** [comment]
Why is there 2 copies of the script?

**Comment by CarlosEsco** [comment]
Who's supposed to run this also? Maybe it can be added to the auto build.  Cause I don't want to run this everytime something is added

**Comment by phanirithvij** [comment]
@CarlosEsco Where is the auto build logic?
I'll add it.

**Comment by arkon** [comment]
https://github.com/inorichi/tachiyomi-extensions/blob/master/.travis.yml

File: scripts/genwiki.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+echo -e "**Page under construction**" >> Extensions.md
+
+for langcode in $(ls src/)
+do 
+    # Language
+    lang=$(lang_from_langcode $langcode)
+    echo Language $lang
+    echo -e "<details open>\n\t<summary>" >> Extensions.md
+    echo -e "<b>$lang</b>" >> Extensions.md
+    echo -e "\t</summary>\n" >> Extensions.md
```
Comment (einstein95) [inline scripts/genwiki.sh]: `\n` isn't needed here and only makes an extra new line
```suggestion
    echo -e "\t</summary>" >> Extensions.md
```

**Comment by phanirithvij** [comment]
@einstein95 here.
[screenshot_2019-02-25-14-43-46-271_com android chrome](https://user-images.githubusercontent.com/29627898/53326549-e15e7c00-390b-11e9-82d2-c3e043ca8516.png)

**Comment by einstein95** [comment]
Ok

**Comment by DitFranXX** [comment]
1. Please add `if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then` ... `fi` before merge this. (for not build on PR)
2. You can't just pull and push right back to repository. IIRC it needs Github personal access token to push. 

or trying to use travis' `deploy` like current `repo` branches.

**Comment by DitFranXX** [comment]
try like this on travis.yml.
```yml
- provider: pages
  local-dir: tachiyomi.wiki
  target-branch: master
  repo: inorichi/tachiyomi-extensions.wiki
  skip-cleanup: true
  keep-history: true
  github-token:
    secure: "SCzQyavQM/bfOjEF+eKp9l0DxiAgsB3v64hYQNb2s/sql4WiL8Mjv9HEAyxBWDfGwUg1cNELWR7s2Y1j/8lCLFaZ5doy7222ZSaRP48MGhpGJQi22jGOqavMNCIcS5EZCjo3OJ9+VWLcx67OIJUQK3KHmnxgeM00luPjzC0a3FiED7/7joU4y9fkQhizfnV6AEZjEma0wUH4+wQxUC+5H71nNDZuz77BO3slZ4TX2YKg9O+QIr5jIjv4GT7dqh+UDQUoMvPXVtORUTI3yOzziwEawc64uMUr+0NtSr9ww+9lg4fs7IeVwkTlDEDbVDFdwvLYAvKlZCqboKygG7NX5Pi11grLYYCrqynXXIsyyBtEyV/fkGt2UcQrMOJRxxI6RkZFuUwozP/DVVO9XiebEfoKKZaQ/ehljveu8+B5S+qsKTO/sFMQJhvi2muhXkUvXeduSjHfqgAxEzrnh6sTR4laI05ICVJVw5j38iUfqOEo4rF+cX11ywgHFzp9RJjYCYhndvobv5vtwk77hrndzVcLlKKlKO4tJDfwg+czRRD7oSX9pT0NeL7tRyD86sNryPMG5pBAfJ4D7knnM9+YmRNJmnsmnu/FGeYlBygUK5IiG05LQ7TfkXiF0wQt7nvlixAiB3ULrso18lqP9U2bGHdjLNaRip3dVxHbrCmP9xY="
  on:
    branch: master
    repo: inorichi/tachiyomi-extensions
```

Tested on https://github.com/DitFranXX/wiki_test_repository/blob/master/.travis.yml
It will works with tweak scripts.

**Comment by phanirithvij** [comment]
@DitFranXX There's already this on `.traivs.yml`

https://github.com/inorichi/tachiyomi-extensions/blob/bf4c14b767824b5a563390c8e57e21b06b27f275/.travis.yml#L31-L41

So what should I do so I could automate the push to the wiki branch?


I need some help.

I know I can't replace the existing logic which uploads apks to the repo named `repo`.
So can I add this below it?


And what exactly does this `if` condition  `if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then ... fi` do?

**Comment by DitFranXX** [comment]
@phanirithvij That's Automated *apk build deploy*. not wiki.

Also, that `if` condition means `Do not generate wiki on pull request`

What i posted is `push to tachiyomi.wiki.git as travis deploy`. normally it can't be push right back using `git` if doesn't provide credentials.

**Comment by fam007e** [comment]
Guys I don't know where to forward this request but I want to know if it is possible to integrate www.8muses.com comics into this tachiyomi app.

**Comment by arkon** [comment]
I'm actually hoping this can be updated to modify _this repo_'s wiki instead, which the main repo's wiki can link to.

@inorichi Would it be possible for you to update permissions for Travis to include the ability to modify the wiki?

**Comment by inorichi** [comment]
I think it's possible now.

**Comment by arkon** [comment]
Superseded by #1348.

**Pull request closed without merging**